estate/model/estate_property.py

```diff
@@ -1,44 +1,181 @@
+            ('sold' , 'Sold') , 
+            ('canceled' , 'Canceled')
+            
+        ] , 
+        required=True  , 
+        copy=False  , 
+        default  = 'new'
+    )
+    
     
+    
+
+    name = fields.Char(
+        required=True,
+    )
+
+    description = fields.Text()
+
     postcode = fields.Char()
 
     date_availability = fields.Date(
-        required=True,
+        copy = False , 
+        default  = _default_vailability()
     )
 
     expected_price = fields.Float()
 
     selling_price = fields.Float(
-        required=True,
+        readonly=True , 
+        copy= False
+        
     )
     
-    bedRooms = fields.Integer()
+    bedRooms = fields.Integer(
+        default  = 2
+    )
     
+
+    bedRooms = fields.Integer()
+
     living_area = fields.Integer()
-    
+
     facades = fields.Integer()
-    
+
     garage = fields.Boolean()
-    
+
     gardern = fields.Boolean()
-    
+
     garden_area = fields.Integer()
-    
+
     gardern_orientation = fields.Selection(
         [
             ("north", "North"),
             ("south", "South"),
             ("east", "West"),
         ]
     )
+
+    property_type_id = fields.Many2one(
+        string='Type',
+        comodel_name='estate.property.types',
+    )
+
+    partner_id = fields.Many2one(
+        string='Buyer',
+        comodel_name='res.partner',
+        copy=False
+    )
+
+    sales_person_id = fields.Many2one(
+        string='Sales Person',
+        comodel_name='res.users',
+        
+        default=lambda self: self.env.user ,
+    )
+    
+    offer_ids =   fields.One2many(
+
+        default=lambda self: self.env.user
+    )
+
+    offer_ids = fields.One2many(
+        string='Offers',
+        comodel_name='estate.property.offer',
+        inverse_name='property_id',
+    )
+
+    tag_ids = fields.Many2many(
+        string='Tags',
+        comodel_name='estate.property.tag',
+    )
+
+    # Computed field
+
+    best_price = fields.Float(compute='_best_price')
+    total_area = fields.Integer(compute='_compute_total_area')
+
+    @api.depends('offer_ids.price')
+    def _best_price(self):
+        for record in self:
+            if record.offer_ids:
+                record.best_price = max(
+                    offer.price for offer in record.offer_ids)
+            else:
+                record.best_price = 0
+
+    @api.depends('living_area', 'garden_area')
+    def _compute_total_area(self):
+
+        for record in self:
+            # Handling null values
+            if record.living_area and record.garden_area:
+                record.total_area = record.living_area + record.garden_area
+            elif record.living_area and not (record.garden_area):
+                record.total_area = record.living_area
+            elif not (record.living_area) and record.garden_area:
+                record.total_area = record.garden_area
+            else:
+                record.total_area = 0
```

Comment:
En Python:
`False + False = 0`
`False + 5 = 5`
```suggestion
            record.total_area = record.living_area + record.garden_area
```